extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/graal/SubstitutionUtils.java

```diff
@@ -0,0 +1,12 @@
+package io.quarkus.kafka.client.runtime.graal;
+
+import java.security.AccessControlContext;
+import java.util.HashMap;
+
+import javax.security.auth.Subject;
+
+public class SubstitutionUtils {
+
+    public static HashMap<AccessControlContext, Subject> subjects = new HashMap<>();
```

Comment:
Should it be `final`?

Reply:
Should this field be moved to the `Subject` class?

Reply:
@cescoffier The field has to be kept outside otherwise IIRC you get an error. The only fields allowed in substituted classes are aliases of existing fields.

I'm not an expert on this java security APIs, but a couple of things to bear in mind would be:

1. Should be it be concurrent?
2. Is there a chance entries could be stored and not removed? Maybe you need some kind of expiration of entries to avoid a leak?

Reply:
ah, thanks for reminding me of this @galderz! I always forgot. 

Handling expiration might be good as it would avoid the potential growth I mentioned in my review.